js/spreadsheet.js

```diff
@@ -67,248 +67,131 @@ var spreadsheet = function(options) {
-   * @param {Object} moveTo - object that returned from closestData() function
+   * The function that surfs through the table data for detecting closest cells with data from the selected cell
    * 
-   * @returns {Array} - coordinats that needs to be selected. Example of the returned data: [[startRow, startCol, endRow, endCol]]
+   * @param {array} tableData - an array of table data received from the hot.getData()
+   * @param {object} cellPosition - an object of the cell coordinates with x and y keys
+   * @param {array} processedCells - an array of the cells we have already checked
+   * @returns {array} returns array of the scaned cells example array[rowIndex][cellIndex] = hasData true/false
    */
-  function coordinatsToSelect(startAt, moveTo) {
-    var firstCol, lastCol, firstRow, lastRow, allData;
-
-    // Returns array of the data from the table with handsontable API
-    allData = hot.getData();
-    startAt = startAt[0];
-
-    if (moveTo.left) {
-      // When data located on the left of the selected cell
-      lastCol = startAt[1];
-
-      // Looking for first col in the array of allData
-      // When we got a null value in the cell it means that we reached the range borders
-      for (var i = lastCol - 1; i >= 0; i--) {
-        if (allData[startAt[0]][i] === null) {
-          firstCol = i;
-          break;
-        }
-      }
-
-      firstCol = firstCol || 0;
-
-      // Looking for the first row in the array of allData
-      // When we got a null value in the cell it means that we reached the range borders
-      for (var i = startAt[0]; i >= 0; i--) {
-        if (allData[i][firstCol] === null) {
-          firstRow = i;
-          break;
-        }
-      }
-
-      firstRow = firstRow || 0;
-
-      // Looking for the last row in the array of allData
-      // When we got a null value in the cell it means that we reached the range borders
-      for (var i = firstRow; i < allData.length; i++) {
-        if (allData[i][firstCol] === null) {
-          lastRow = i;
-          break;
-        }
-      }
-
-      lastRow = _.max([lastRow - 1, 0]);
-    } else if (moveTo.right) {
-      // When data located on the right of the selected cell
-      firstCol = startAt[1];
-
-      for (var i = firstCol + 1; i < allData.length; i++) {
-        if (allData[startAt[0]][i] === null) {
-          lastCol = i - 1;
-          break;
-        }
-      }
+  function getDataCoordinates(tableData, cellPosition, processedCells) {
+    if (!processedCells) {
+      processedCells = [];
+      processedCells[cellPosition.x] = [];
+      processedCells[cellPosition.x][cellPosition.y] = true;
+    }
 
-      for (var i = startAt[0]; i > 0; i--) {
-        if (allData[i][lastCol] === null) {
-          firstRow = i ? i - 1 : i;
-        }
-      }
+    var cellsGroup = getNearestCells(cellPosition);
 
-      firstRow = firstRow || 0;
+    cellsGroup['currentCell'] = cellPosition;
```

Comment:
Why would there be a `currentCell` property? I don't see it in the `getNearestCells()` function.

Reply:
@tonytlwu 
`currentCell` needed for when our selected cell is without a data on it row or col.

Reply:
@YaroslavOvdii Let's just add it into `getNearestCells()`. The `currentCell` property is 100% always added at this point, and it's good for the nearest cells to have a reference of which cell was the starting point.